sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/PropertiesAccessCorrectFieldBase.cs

```diff
@@ -83,16 +84,21 @@ protected void CheckType(SymbolAnalysisContext context)
                 .OfType<IPropertySymbol>()
                 .Where(p => !p.IsImplicitlyDeclared);
 
-        protected void CheckExpectedFieldIsUsed(IFieldSymbol expectedField, FieldData? actualField, SymbolAnalysisContext context)
+        protected void CheckExpectedFieldIsUsed(IFieldSymbol expectedField, IEnumerable<FieldData?> actualFields, SymbolAnalysisContext context)
         {
-            if (actualField.HasValue && actualField.Value.Field != expectedField)
+            var expectedFieldIsUsed = actualFields.Any(a => a.HasValue && a.Value.Field == expectedField);
+            if (!expectedFieldIsUsed)
             {
-                context.ReportDiagnosticWhenActive(Diagnostic.Create(
-                    Rule,
-                    actualField.Value.LocationNode.GetLocation(),
-                    actualField.Value.AccessorKind == AccessorKind.Getter ? "getter" : "setter",
-                    expectedField.Name
-                    ));
+                var fieldWithValue = actualFields.FirstOrDefault(a => a.HasValue);
```

Comment:
I'm guessing it's a deliberate choice to report on the first field usage in the event of the expected field not being used, but it's not clear. It's worth calling out in a comment if this is the case.